### PR TITLE
Allow flux_subprocess_kill() on remote processes not yet running

### PR DESCRIPTION
**Pull request by grondo**
This PR is somewhat related to #2275.

When signalling a set of remote processes, it is inconvenient to handle the case of processes for which a request has been sent, but a response has not yet been received. Currently, `flux_subprocess_kill(3)` will return EINVAL on these processes since they are not in the RUNNING state.

Instead of requiring the libsubprocess user to handle this case by tracking which remote processes have pending signals, this PR adds a `signal_pending` flag for remote subprocesses which tries to do the right thing.

If a user calls `flux_subprocess_kill` on a remote process which is in the INIT state, then the pending flag is set and an empty future is returned to the caller. Later, if that subprocess reaches the RUNNING state, *then* the signal is forwarded, and its response fulfills the future returned to the caller earlier.

If the subprocess fails then the future is fulfilled with an error.

For testing new options were added to the `t/rexec/rexec` test program, which then obviated the need for the signal specific `t/rexec/rexec_signal`, which was subsequently removed.

All this will make it much easier for the bulk exec code in the job-exec module to cancel a job due to an exception when the job is in the middle of starting up.

### CODE REVIEW

**Comment by garlick** [comment]
Nice coverage!  This looks OK to me although I admit I don't understand the code very well.  Probably be good to get a second look by @chu11.

File: t/rexec/rexec.c [hunk 1]
```diff
@@ -40,11 +44,32 @@ int exit_code = 0;
...
+void kill_cb (flux_future_t *f, void *arg)
+{
+    if (flux_future_get (f, NULL) < 0)
+        log_err ("kill_cb: flux_subprocess_kill");
+    flux_future_destroy (f);
+}
+
+void send_sigterm (flux_subprocess_t *p)
+{
+    flux_future_t *f = flux_subprocess_kill (p, 15);
```
[the remark this answers is not in the document]
Comment (grondo) [reply t/rexec/rexec.c]: I think it was just to avoid pulling in signal.h. Silly reason, I'll fix it.

**Comment by grondo** [comment]
Ok, fixed that minor nit and pushed with a rebase.

**Comment by chu11** [comment]
awesome, thanks!

**Comment by grondo** [comment]
Thanks!

**Pull request merged**